browser/brave_content_browser_client_browsertest.cc

```diff
@@ -483,6 +485,40 @@ IN_PROC_BROWSER_TEST_F(BraveContentBrowserClientTest,
       << "No changes on the real URL";
 }
 
+IN_PROC_BROWSER_TEST_F(BraveContentBrowserClientTest, MixedContentForOnion) {
+  // Don't block the mock .onion requests.
+  tor::OnionLocationNavigationThrottle::BlockOnionRequestsOutsideTorForTesting(
+      false);
+
+  const GURL onion_url =
+      embedded_test_server()->GetURL("test.onion", "/simple.html");
+  ASSERT_EQ("http", onion_url.scheme());
+  ASSERT_TRUE(ui_test_utils::NavigateToURL(browser(), onion_url));
+  content::WebContents* contents =
+      browser()->tab_strip_model()->GetActiveWebContents();
+  WaitForLoadStop(contents);
+  {
+    content::WebContentsConsoleObserver console_observer(contents);
+    console_observer.SetPattern(
+        "Mixed Content: The page at 'http://test.onion*/simple.html' was "
+        "loaded over HTTPS, but requested an insecure resource "
+        "'http://example.com/'. This request has been blocked; the content "
+        "must be served over HTTPS.");
+    ASSERT_FALSE(content::ExecJs(contents, "fetch('http://example.com')"));
+    console_observer.Wait();
+  }
+  {
+    content::WebContentsConsoleObserver console_observer(contents);
+    ASSERT_FALSE(content::ExecJs(contents, "fetch('https://example.com')"));
+    EXPECT_TRUE(console_observer.messages().empty());
+  }
+  {
+    content::WebContentsConsoleObserver console_observer(contents);
+    ASSERT_FALSE(content::ExecJs(contents, "fetch('https://example.onion')"));
+    EXPECT_TRUE(console_observer.messages().empty());
+  }
+}
```

Comment:
Can we add a test for `fetch('http://example.onion')` which should be **allowed** since it's actually a secure URL despite the HTTP scheme?